src/main/java/org/opensearch/ad/task/ADTaskCacheManager.java

```diff
@@ -577,9 +612,142 @@ public Integer getTopEntityCount(String detectorId) {
      * @param allowedRunningEntities max allowed running entities
      */
     public void setAllowedRunningEntities(String detectorId, int allowedRunningEntities) {
+        logger.debug("Set allowed running entities of detector {} as {}", detectorId, allowedRunningEntities);
         getExistingHCTaskCache(detectorId).setEntityTaskLanes(allowedRunningEntities);
     }
 
+    /**
+     * Set detector task slots. We cache task slots assigned to detector on coordinating node.
+     * When start new historical analysis, will gather detector task slots on all nodes and
+     * check how many task slots available for new historical analysis.
+     *
+     * @param detectorId detector id
+     * @param taskSlots task slots
+     */
+    public synchronized void setDetectorTaskSlots(String detectorId, int taskSlots) {
+        logger.debug("Set task slots of detector {} as {}", detectorId, taskSlots);
+        ADTaskSlotLimit adTaskSlotLimit = detectorTaskSlotLimit
+            .computeIfAbsent(detectorId, key -> new ADTaskSlotLimit(taskSlots, taskSlots));
+        adTaskSlotLimit.setDetectorTaskSlots(taskSlots);
+    }
+
+    /**
+     * Scale up detector task slots.
+     * @param detectorId detector id
+     * @param delta scale delta
+     */
+    public synchronized void scaleUpDetectorTaskSlots(String detectorId, int delta) {
```

Comment:
Maybe add some error handling or logging for failure, if the scaling up failed? Same goes for `scaleDownHCDetectorTaskSlots`

Reply:
I think it's ok to keep as it as 
1. These two functions `scaleUpDetectorTaskSlots` and `scaleDownHCDetectorTaskSlots` is straight forward and simply set value in cache.
2. If any exception, the client using them will handle.